packages/react/src/primitives/Checkbox/Checkbox.tsx

```diff
@@ -50,9 +50,11 @@ const CheckboxPrimitive: Primitive<CheckboxProps, 'input'> = (
   const dataId = useStableId();
   React.useEffect(() => {
     const input = document.querySelector(`[data-id="${dataId}"]`);
-    // HTMLInputElement does not have an `indeterminate` attribute
-    (input as HTMLInputElement & { indeterminate: boolean }).indeterminate =
-      isIndeterminate;
+    if (input && isIndeterminate !== undefined) {
```

Comment:
Probably not super important but just for explicitness should the second condition be `typeof isIndeterminate === 'boolean'`?

Reply:
That does seem clearer, will update